### DISCUSSION TITLE
Ready features still missing in react-navigation/native-stack, like the fullscreen swipe

### DISCUSSION THREAD

**Discussion by hirbod** [software-mansion/react-native-screens#1293] [Q&A] [answered]
@WoLewicki who is responsible for features like the fullscreen swipe to be available in @react-navigation/native-stack? I mean this feature is ready for a while now and since you guys shifted native-stack back to react-navigation, those new props haven't been integrated yet and there is no response on the canny for it. Importing native-stack from RNS though introduces a lot of TypeScript errors for me. How can we help to make changes made here to be pushed on time to react navigation?

**Comment by kacperkapusciak** [comment, marked as answer]
Hi @hirbod :wave:

I guess I am responsible for it and I'm sorry for the lack of transparency.

There were a bunch of factors that slowed the integration of features to react-navigation. To name a few:
- what can and cannot be integrated to react-navigation is tightly dependent on Expo's release cycle
- the main maintainer of react-navigation didn't have time and resources to work on the library so these changes wouldn't be released
- and this one is on me - I got sidetracked by different projects

I've sent the PR adding the full-screen swipe to the react-navigation repo. I hope it's getting released soon after it's being merged.

Cheers

**Comment by hirbod** [comment]
Thank you very much @kacperkapusciak 

You don't need to justify, I hope you didn't felt offended, I was just curious because I wanted to poke into the right direction.
Thanks a lot for clarification.

**Comment by hirbod** [comment]
I've just tested the brand new react-navigation release and the fullScreenGestureEnabled + customAnimation is just freaking awesome. And finally we have animated appear/disappear of the keyboard. Thats so nice! Thanks a ton @kacperkapusciak

**Reply by Chasty** [reply]
@hirbod is this already in react-navigation? I can't do the fullscreen swipe yet.

**Reply by Chasty** [reply]
for me the `fullScreenGestureEnabled + customAnimation` is not working : (

```
<ModalStack.Screen
              options={{
                //headerShown: false,
                animation: 'simple_push',
                fullScreenGestureEnabled: true,
                gestureEnabled: true,
                customAnimationOnGesture: true,
                presentation: 'transparentModal',
              }}
              name={ModalNavigationScreen.PageViewer}
              component={PageViewer}
            />
```

**Reply by hirbod** [reply]
fullScreenGestureEnabled does not work on transparentModal.

If you want a modal style animation, you need to set customAnimation: "slide_from_bottom" and gestureDirection: "vertical" for the card presentation. But you can't set them right now, only for normal cards, since my PR hasn't been merged yet 

https://github.com/react-navigation/react-navigation/pull/10578

**Reply by Chasty** [reply]
Thanks for you quick reply, `customAnimation` for me is `animation` I don't have `gestureDirection` prop yet, so I assume your PR gonna add this one, for me it is working just with `modal` presentation not with `card` presentation to do the swipe.

**Reply by hirbod** [reply]
Yeah, sorry. animation "slide_from_bottom" will be enough to mimic the modal. 
My PR adds automatic handling for it, so you don't even have to set anything beside "gestureDirection"

https://github.com/react-navigation/react-navigation/pull/10578/files#diff-430ad7ad5c4108117dc2b61f532413f1f1a8d9766b1f9554d64879a9a3f2610fR155-R170

**Reply by Chasty** [reply]
@hirbod we have created a patch from your PR, but it seems that the swipe down gesture not working,

```
 <ModalStack.Screen
              options={{
                headerShown: false,
                presentation: 'card',
                gestureDirection: 'vertical',
              }}
              name={ModalNavigationScreen.PageViewer}
              component={PageViewer}
      />
```

check the video here.

https://drive.google.com/file/d/1O_-25wpathJyA0IxISJq5TZxzSV-rmKp/view?usp=sharing

and this is the patch

https://pastebin.com/WxggUTEr

**Reply by hirbod** [reply]
you need to also set customAnimationOnSwipe to true

**Comment by adampbf** [comment]
@hirbod  is it possible to merge this PR in sooner future ?

**Reply by hirbod** [reply]
I am no member, just a contributor. I have no power to merge it.
I am also waiting for the merge since 20 days..

**Comment by Chasty** [comment]
so, maybe @kacperkapusciak can help with that.

**Reply by hirbod** [reply]
No
